File: Iteration2/code/CSE3063GRP12/src/features/advisor/approval_courses_selected/ApprovalCoursesSelectedView.java
package features.advisor.approval_courses_selected;

import java.util.ArrayList;
import core.models.concretes.Course;

public class ApprovalCoursesSelectedView {

	public void showSelectedCourses(ArrayList<Course> selectedCourseList) {
    System.out.println("====================================================|| Registration Management ||====================================================");
    System.out.println("###======List of Student Selected Courses:");
    System.out.println("=====================================================================================================================================");
    System.out.printf("%-3s\t%-20s %-50s %-8s %-8s %-8s\n", " No.", "Course Code", "Course Name", "Quota", "Semester No.", "ECTS");
    System.out.println("=====================================================================================================================================");

    for (int i = 0; i < selectedCourseList.size(); i++) {
        Course currentCourse = selectedCourseList.get(i);
        System.out.printf(" [%-1d]\t%-20s %-52s %-12d %-8d %-8d\n",
        i + 1, currentCourse.getCourseCode(), currentCourse.getName(), 
		currentCourse.getQuota(), currentCourse.getSemester(), currentCourse.getCredit());
    }

    System.out.println("=====================================================================================================================================");
	}

	public void showSuccessMessage() {
		System.out.println("You have successfully approved the course list");
	}

	public void showErrorMessage(Exception exception) {
		System.err.println(exception);
	}
}

File: Iteration3/code/CSE3063GRP12/src/features/advisor/approval_courses_selected/ApprovalCoursesSelectedView.py
class ApprovalCoursesSelectedView:

    def show_selected_courses(self,selected_course_list):
        print("====================================================|| Registration Management ||====================================================")
        print("###======List of Student Selected Courses:")
        print("=====================================================================================================================================")
        print("{:<3s}\t{:<20s} {:<50s} {:<8s} {:<10s}  {:<8s}".format("No.", "Course Code", "Course Name", "Quota", "Semester No.", "ECTS"))
        print("=====================================================================================================================================")

        for i, current_course in enumerate(selected_course_list):
            print(" [{:<1d}]\t{:<20s} {:<52s} {:<12d} {:<8d} {:<8d}".format(
                i + 1, current_course['courseCode'], current_course['name'], 
                current_course['quota'], current_course['semester'], current_course['credit']))

        print("=====================================================================================================================================")

    
    def show_success_message():
        print("You have successfully approved the course list")

  
    def show_error_message(self,exception):
        print("Error: " + str(exception))